libraries/abi_generator/abi_generator.cpp

```diff
@@ -573,6 +603,29 @@ string abi_generator::add_vector(const clang::QualType& vqt, size_t recursion_de
   return vector_element_type_str;
 }
 
+string abi_generator::add_fixed_array(const clang::QualType& vqt, size_t recursion_depth) {
+
+  ABI_ASSERT( ++recursion_depth < max_recursion_depth, "recursive definition, max_recursion_depth" );
```

Comment:
I think the error would be clearer if the message was "recursive definition, exceeded recursive depth of ${depth}", (max_recursion_depth)

Reply:
ok. will modify it accordingly.